addons/product/controllers/product_configurator.py

```diff
@@ -0,0 +1,226 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import Controller, request, route
+
+
+class ProductConfiguratorController(Controller):
+
+    @route('/product_configurator/get_values', type='json', auth='user')
+    def get_product_configurator_values(
+        self,
+        product_template_id,
+        company_id=None,
+        ptav_ids=None,
+        **kwargs
+    ):
+        """ Return all product information needed for the product configurator.
+
+        :param int product_template_id: The product for which to seek information, as a
+                                        `product.template` id.
+        :param int|None company_id: The company to use, as a `res.company` id.
+        :param recordset|None ptav_ids: The combination of the product, as a
+                                        `product.template.attribute.value` recordset.
+        :param dict kwargs: Optional data passed to helper methods.
+        :rtype: dict
+        :return: A dict containing a list of products information, generated by
+                 :meth:`_get_product_information`.
+        """
+        if company_id:
+            request.update_context(allowed_company_ids=[company_id])
+        product_template = request.env['product.template'].browse(product_template_id)
+
+        combination = request.env['product.template.attribute.value']
+        if ptav_ids:
+            combination = request.env['product.template.attribute.value'].browse(ptav_ids).filtered(
+                lambda ptav: ptav.product_tmpl_id.id == product_template_id
+            )
+        if not combination:
+            combination = product_template._get_first_possible_combination()
+
+        return {
+            "products": [
+                dict(
+                    **self._get_product_information(
+                        product_template,
+                        combination,
+                        **kwargs
+                    ),
+                    parent_product_tmpl_ids=[],
```

Comment:
This has to stay in `sale_product_configurator`, there is no optional nor parent product in `product`.